pkg/apis/eksctl.io/v1alpha5/identitymapping.go

```diff
@@ -0,0 +1,56 @@
+package v1alpha5
+
+import (
+	"github.com/pkg/errors"
+)
+
+// IAMIdentityMapping contains IAM accounts, users, roles and services that will be added to the
+// aws-auth configmap to enable access to the cluster
+type IAMIdentityMapping struct {
+	// +optional
+	ARN             string   `json:"arn,omitempty"`
+	Username        string   `json:"username,omitempty"`
+	Groups          []string `json:"groups,omitempty"`
+	Account         string   `json:"account,omitempty"`
+	ServiceName     string   `json:"serviceName,omitempty"`
+	Namespace       string   `json:"namespace,omitempty"`
+	NoDuplicateArns bool     `json:"noDuplicateArns,omitempty"`
+}
+
+func (im *IAMIdentityMapping) hasARNOptions() bool {
+	return (im.hasARN() && im.hasUsername() && im.hasGroups())
```

Comment:
Nit: superfluous parentheses. 
```suggestion
	return im.hasARN() && im.hasUsername() && im.hasGroups()
```